index.html

```diff
@@ -2530,6 +2530,78 @@ <h5>Additional Properties in the Manifest</h5>
+			<ol>
+				<li>(<a href="#wptitle"></a>) if <code>manifest["name"]</code> is <code>undefined</code>, then locate the <a href="https://www.w3.org/TR/html/document-metadata.html#the-title-element"><code>title</code></a> HTML element&nbsp;[[!html]] using <code>document</code>. If that element exists, let <code>t</code> be its text content, and add to <code>manifest</code>:
+					<ul>
+						<li>if the language of <code>title</code> is explicitly <a href="https://www.w3.org/TR/html/dom.html#the-lang-and-xmllang-attributes">set</a> to the value of <code>l</code>, then add <br/><code>"name" : {"@value" : t, "@language" : l}</code><br/></li>
+						<li>or <br/><code>"name" : t</code><br> otherwise</li>
+					</ul>
+				</li>
+				<li>(<a href="#language-and-dir"></a>) if <code>manifest["inLanguage"]</code> is <code>undefined</code> and the value of <code>lang</code> is <em>not</em> <code>undefined</code>, add<br><code>"inLanguage" : lang</code><br> to <code>manifest</code> </li>
+				<li>(<a href="#language-and-dir"></a>) if <code>manifest["inDirection"]</code> is <code>undefined</code> and the value of <code>dir</code> is <em>not</em> <code>undefined</code>, add<br><code>"inDirection" : dir</code><br> to <code>manifest</code> </li>
+				<li>(<a href="#arrays-and-single-values"></a>) if the value of <code>P["term"]</code>, where <code>P</code> is any object in <code>manifest</code> (including itself) and <code>term</code> is:
+					<ul>
+						<li><code>@type</code>; or</li>
+						<li>one of the <a href="#accessibility">accessibility</a> terms except for <code>accessibilitySummary</code>; or</li>
+						<li>one of the <a href="#creators">creator</a> terms; or</li>
+						<li><code>name</code>; or</li>
+						<li>one of the <a href="#resource-categorization-properties">resource categorization properties</a>; or</li>
+						<li><code>rel</code></li>
+					</ul>
+					is a single string or object <code>v</code>, then change the relevant term/value to<br><code>"term" : [v]</code>
+				</li>
+				<li>(<a href="#creators"></a>) if one of values <code>v</code> in the <code>manifest["term"]</code> array, where <code>term</code> is one of the <a href="#creators">creator</a> terms, is a single string, exchange that element of the array to<br><code>"term" : { "@type": "Person", "name" : [v]}</code></li>
+				<li>(<a href="#link-values"></a>) if one of values <code>v</code> in the <code>manifest["term"]</code> array, where <code>term</code> is one of the <a href="#resource-categorization-properties">resource categorization properties</a>, is a single string, exchange that element of the array to<br><code>"term" : { "@type": "PublicationLink", "url" : v}</code><br></li>
```

Comment:
The question what we do with @type when we map it against WebIDL. After all, we do have a PublicationLink interface, and the mapping of @type is, in some sense, mapping the object on the right interface in the WebIDL.

I would leave it as is for now, and come back to this when we discuss the details of the next step.

Reply:
From a WebIDL perspective:
- having a `@type` for each publication link is fairly useless (we know that we have a sequence of PublicationLink anyway)
- we can't have a Person and an Organization dictionary if we plan on doing a straight conversion from JSON to WebIDL using the canonical manifest

Reply:
This is the reference used in the Web App Manifest draft: https://heycam.github.io/webidl/#es-dictionary

Reply:
@HadrienGardeur .

> This is the reference used in the Web App Manifest draft: https://heycam.github.io/webidl/#es-dictionary

... which is, in fact, a strange reference because it describes an IDL->JavaScript mapping, not a JSON->IDL mapping that is used here. But o.k., it describes an equivalence of IDL and JS data, I guess it is acceptable.

I guess what we have to look at, in view of this mapping, is 

1. what we do with `@type` and `@id`, which are odd-man-out because they can be problematic as attribute names (I have not found a clear specification of the IDL term syntax, but I realized that the IDL parser working in respec simply swallows the `@` sign when generating the final text.)
2. What to do with the Person/Organization IDL interfaces.

To move ahead, maybe we should (after discussions today) agree to merge this PR with the canonical manifest as a concept and its current spec (ie, making it part of the current editor's draft) and then solve these two issues to update the draft. We can then look at the details of the conversion which, I believe, should simply list the type of error/warning checks that the processor should do no the final data. W.D.Y.T?

Reply:
For Person/Organization, I'm not convinced that we need two separate IDL interfaces. A `type` for a universal interface would work as well since they'll contain the same terms anyway.

Reply:
@HadrienGardeur, mainly if we follow this very close relationship between the canonical manifest and IDL, I agree with you. I would put 'type' required for contributors, and use that mechanism.

I think we should also map, via the context file, `@value`->`@value`, `@language` -> `language`, `@id` -> `id`, and `@type` -> `type`. We can do that, it seems that the google tools understand them, and it makes the mapping from the can. manifest to IDL easy and problem-less.

But we can do that only if we have a general WG agreement on the can. manifest, because it is a change affecting the whole manifest definition.

Reply:
From an IDL standpoint, the type for Person/Organization isn't really required, as it won't make much of a difference for a UA if it's one or the other. Can't we default to Person?

Reply:
The schema.org requirements means that some creators must be persons only, others can be either persons or organizations. If the typing is not there, schema.org consumers might reject properties (I think they do it already), meaning that, depending on the creators, these constraints must be checked.